staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2beta1/api.pb.go

```diff
@@ -104,28 +104,28 @@ func (m *EncryptedObject) GetAnnotations() map[string][]byte {
 }
 
 func init() {
-	proto.RegisterType((*EncryptedObject)(nil), "v2alpha1.EncryptedObject")
-	proto.RegisterMapType((map[string][]byte)(nil), "v2alpha1.EncryptedObject.AnnotationsEntry")
+	proto.RegisterType((*EncryptedObject)(nil), "v2beta1.EncryptedObject")
+	proto.RegisterMapType((map[string][]byte)(nil), "v2beta1.EncryptedObject.AnnotationsEntry")
```

Comment:
when we graduate to "v2", will we have to duplicate the proto packages and maintain both v2beta1 and v2 types and convert between them ~forever? if so, should we just name the package and types `v2` today, and limit the beta-ness to the feature gate, string value of StatusResponse.version, and apiVersion in the encryption config file entry?

that would mean we'd be limited to making forward compatible changes to the proto (only add fields), which I think we'd already have to do, and would have signal in the encryption config file and StatusResponse if the user and kms plugin expected to have beta semantics or GA semantics

Reply:
> when we graduate to "v2", will we have to duplicate the proto packages and maintain both v2beta1 and v2 types and convert between them ~forever? if so, should we just name the package and types v2 today, and limit the beta-ness to the feature gate, string value of StatusResponse.version, and apiVersion in the encryption config file entry?

we can do this! @enj had the similar comment as well.

Reply:
> when we graduate to "v2", will we have to duplicate the proto packages and maintain both v2beta1 and v2 types and convert between them ~forever? if so, should we just name the package and types `v2` today, and limit the beta-ness to the feature gate, string value of StatusResponse.version

+1

> and apiVersion in the encryption config file entry?

This is already set to `v2` today and doesn't have the alpha and beta concept.

> that would mean we'd be limited to making forward compatible changes to the proto (only add fields), which I think we'd already have to do

+1

> would have signal in the encryption config file and StatusResponse if the user and kms plugin expected to have beta semantics or GA semantics

When we go GA, could we just probe `StatusResponse.version` on start up / automatic config reload and pin the plugin's version to the first response?